spec/requests/vms_spec.rb

```diff
@@ -1117,6 +1117,22 @@ def update_raw_power_state(state, *vms)
           expect(response.parsed_body).to include(expected)
         end
 
+        it "creates a request from VmOrTemplate" do
+          api_basic_authorize(action_identifier(:vms, :request_retire))
+          message = "VM id:#{vm.id} name:'#{vm.name}' request retire"
+          task_id = MiqTask.find_by(:name => message)&.id
+          expect(task_id).to be_nil
+          expect(MiqRequest.count).to eq(0)
+
+          post(vm_url, :params => gen_request(:request_retire))
+
+          task = MiqTask.find_by(:name => message)
+
+          MiqTask.find(task.id).miq_queue.deliver
+
+          expect(VmOrTemplate).to receive(:make_retire_request).once
```

Comment:
Is this actually `expect_any_instance_of(VmOrTemplate)`?

Reply:
Yes. It needs to be calling https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_or_template.rb#L351.